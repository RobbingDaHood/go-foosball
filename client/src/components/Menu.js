import React from 'react';
import PropTypes from 'prop-types';
import AppBar from 'material-ui/AppBar';
import IconButton from 'material-ui/IconButton';
import HomeIcon from 'material-ui-icons/Home';
import RefreshIcon from 'material-ui-icons/Refresh';
import Typography from 'material-ui/Typography';
import Toolbar from 'material-ui/Toolbar';
import { withStyles } from 'material-ui/styles';
import withRoot from '../withRoot';
import { Link } from 'react-router-dom';

const styles = theme => ({
  flex: {
    flex: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
});

class Menu extends React.Component {
  render() {
    const { classes } = this.props;
    return (
      <AppBar position="static">
        <Toolbar>
          <IconButton
            to="/"
            component={props => <Link {...props} />}
            className={classes.menuButton}
            color="inherit"
            aria-label="Home"
          >
            <HomeIcon />
          </IconButton>
          <Typography variant="title" color="inherit" className={classes.flex}>
            {this.props.title}
          </Typography>
          {this.props.fetch ? (
            <IconButton
              color="inherit"
              aria-label="Refresh"
              onClick={this.props.fetch}
            >
              <RefreshIcon />
            </IconButton>
          ) : null}
        </Toolbar>
      </AppBar>
    );
  }
}

Menu.propTypes = {
  classes: PropTypes.object.isRequired,
  title: PropTypes.string.isRequired,
  fetch: PropTypes.func,
};

export default withRoot(withStyles(styles)(Menu));
