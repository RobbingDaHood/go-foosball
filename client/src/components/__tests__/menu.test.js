// Link.react.test.js
import React from 'react';
import Menu from '../Menu';
import renderer from 'react-test-renderer';

test('Should render with title', () => {
  const component = renderer.create(<Menu title="test" />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
