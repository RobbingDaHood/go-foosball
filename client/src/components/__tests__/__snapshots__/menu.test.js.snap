// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render with title 1`] = `
<header
  className="MuiPaper-root-12 MuiPaper-elevation4-18 MuiAppBar-root-4 MuiAppBar-positionStatic-8 MuiAppBar-colorPrimary-10"
>
  <div
    className="MuiToolbar-root-39 MuiToolbar-gutters-40"
  >
    <a
      aria-label="Home"
      className="MuiButtonBase-root-47 MuiIconButton-root-41 MuiIconButton-colorInherit-42 Menu-menuButton-2"
      href="/"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex="0"
    >
      <span
        className="MuiIconButton-label-46"
      >
        <svg
          aria-hidden="true"
          className="MuiSvgIcon-root-50"
          color={undefined}
          focusable="false"
          viewBox="0 0 24 24"
        >
          <g>
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </g>
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root-56"
      />
    </a>
    <h2
      className="MuiTypography-root-63 MuiTypography-title-69 MuiTypography-colorInherit-82 Menu-flex-1"
    >
      test
    </h2>
    <div
      className="Menu-buttons-3"
    />
  </div>
</header>
`;
